src/components/MdTable/MdTablePagination.vue

```diff
@@ -4,86 +4,171 @@
+            } else {
+              // internal pagination
+              this.updatePage()
+            }
+          }
         }
       }
     },
     methods: {
-      setPageSize () {
-        this.$emit('update:mdPageSize', this.currentPageSize)
+      getPageCount () {
+        if (this.mdCount % this.currentPageSize) {
+          return Math.floor((this.mdCount / this.currentPageSize)) + 1
+        } else {
+          return Math.floor((this.mdCount / this.currentPageSize))
+        }
+      },
+      setPage (mdPage, mdCount) {
+        this.mdPage = mdCount > 0 ? (mdPage > 0 ? mdPage : (this.mdPage > 0 ? this.mdPage : 1)) : 0
+        var pageCount = this.getPageCount()
+        if (this.mdPage > pageCount) {
+          this.mdPage = pageCount
+        }
+      },
+      updatePage () {
+        if (this.currentPageSize !== 0) {
+          if (this.mdData && !this.mdData.mdData) {
+            // internal pagination
+            this.mdCount = this.mdData.length
+            this.setPage(this.mdPage, this.mdData.length)
+            if (this.mdPage > 0) {
+              this.$emit('update:mdPaginatedData', getPageData(this.mdData, this.mdPage, this.currentPageSize))
+            } else {
+              this.$emit('update:mdPaginatedData', [])
+            }
+          } else if (this.mdData) {
+            // external pagination
+            this.mdCount = this.mdData.mdCount
+            this.setPage(this.mdData.mdPage, this.mdCount)
+            this.$emit('update:mdPaginatedData', this.mdData.mdData)
+          } else {
+            this.mdCount = 0
+            this.mdPage = 0
+          }
+        }
       },
       goToPrevious () {
-
+        if (this.mdData && this.mdData.mdData) {
+          // external pagination/sort
+          if (this.mdUpdate(this.mdPage - 1, this.currentPageSize, this.MdTable.sort, this.MdTable.sortOrder) !== false) {
+            this.mdPage = this.mdPage - 1
+          }
+        } else {
+          // internal pagination/sort
+          this.mdPage = this.mdPage - 1
+          this.$emit('update:mdPaginatedData', getPageData(this.mdData, this.mdPage, this.currentPageSize))
+        }
       },
       goToNext () {
-
+        if (this.mdData && this.mdData.mdData) {
+          // external pagination/sort
+          if (this.mdUpdate(this.mdPage + 1, this.currentPageSize, this.MdTable.sort, this.MdTable.sortOrder) !== false) {
+            this.mdPage = this.mdPage + 1
+          }
+        } else {
+          // internal pagination/sort
+          this.mdPage = this.mdPage + 1
+          this.$emit('update:mdPaginatedData', getPageData(this.mdData, this.mdPage, this.currentPageSize))
+        }
+      },
+      sort () {
+        if (this.mdData && this.mdData.mdData) {
+          // external pagination/sort
+          this.mdUpdate(this.mdPage, this.currentPageSize, this.MdTable.sort, this.MdTable.sortOrder)
```

Comment:
Some of this logic seems similar across `goToPrevious`, `goToNext` and `sort`.
Do you think it would be possible to group them together into a generic helper?